### DISCUSSION TITLE
How to use debounce with `$watch`?

### DISCUSSION THREAD

**Discussion by canvural** [alpinejs/alpine#964] [1. Help]
I have an item in my data, and every time it updates I will do a `fetch`. It is updated outside of Alpine. I don't want to fire the request on every character, so I want to use debounce.

But there is no example for it that I could find. And I tried many different ways, but can't seem to get it working correctly.

Anyone knows a good way to achieve this?

**Comment by ryangjchandler** [comment]
@canvural You can find info about debouncing with `x-model` here: https://github.com/alpinejs/alpine#x-model

**Comment by canvural** [comment]
@ryangjchandler I'm sorry, I wrote the issue not correctly. I don't use `x-model`

My setup is, I have an editor with CodeMirror, and when the text in the editor changes I'm updating the Alpine's data from outside. Here is the relevant snippet:

```js
...
const code = document.getElementById('code');

  codeMirror.getDoc().on('change', function (cm, _) {
    code.__x.$data.setCode(cm.getValue())
  });
...
```

`setCode` is also my custom function. So the function for Alpine's `x-data` is something like this:
```js
function data() {
  return {
    code: '',
    setCode(newCode) { this.code = newCode }
  }
}
```

So I want to watch the `code` for changes, but not for every character. So I need to debounce.

**Reply by SimoTod** [reply]
There's no standalone debounce in Alpine. Unfortunately you need to implement it by yourself. If you google "debounce javascript" you should be able to find the relevant code.
